src/objects/BatchedMesh.js

```diff
@@ -868,6 +868,19 @@ class BatchedMesh extends Mesh {
 
 	}
 
+	getGeometryRangeAt( geometryId ) {
+
+		if ( geometryId < 0 || geometryId >= this._geometryCount ) {
+
+			return null;
+
+		}
+
+		const drawRange = this._drawRanges[ geometryId ];
+		return { ...drawRange }; // cloned to avoid external changes
```

Comment:
Is it planned to use this method per animation step?

If so, the policy is to provide a `target` object and not to clone/create objects.

Reply:
I'm happy with either. I don't expect this to be the kind of thing that will be called every frame but I suppose it could be. As mentioned our use case is for [generating BVH structures](https://github.com/mrdoob/three.js/pull/29409#issuecomment-2352753013) for the underlying geometry - which is an inherently expensive operation already (in terms of memory and time).

Reply:
In this case, a target object that holds the copied draw range data sounds more appropriate to me.

Reply:
I corrected it and updated the doc :)